File: app/src/main/java/com/rodolfonavalon/canadatransit/model/transit/Meta.java
package com.rodolfonavalon.canadatransit.model.transit;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import lombok.Getter;

public class Meta {

    @SerializedName("sort_key")
    @Getter private String sortKey;
    @SerializedName("sort_order")
    @Getter private String sortOrder;

    @Nullable
    @SerializedName("next")
    @Getter private String next;
    @Nullable
    @SerializedName("prev")
    @Getter private String prev;

    @SerializedName("per_page")
    @Getter private int perPage;
    @SerializedName("offset")
    @Getter private int offset;

    public boolean hasNext() {
        return next != null;
    }
}

File: app/src/main/kotlin/com/rodolfonavalon/canadatransit/model/transit/Meta.kt
package com.rodolfonavalon.canadatransit.model.transit

import com.squareup.moshi.Json

class Meta(
    @field:Json(name = "sort_key") val sortKey: String,
    @field:Json(name = "sort_order") val sortOrder: String,
    @field:Json(name = "next") val next: String?,
    @field:Json(name = "prev") val prev: String?,
    @field:Json(name = "per_page") val perPage: Int,
    @field:Json(name = "offset") val offset: Int
) {

    operator fun hasNext(): Boolean {
        return next != null
    }
}
